search-service/src/main/kotlin/com/egm/stellio/search/service/TemporalEntityService.kt

```diff
@@ -0,0 +1,94 @@
+package com.egm.stellio.search.service
+
+import com.egm.stellio.search.model.AttributeInstanceResult
+import com.egm.stellio.search.model.TemporalEntityAttribute
+import com.egm.stellio.shared.model.CompactedJsonLdAttribute
+import com.egm.stellio.shared.model.CompactedJsonLdEntity
+import com.egm.stellio.shared.util.*
+import org.springframework.stereotype.Service
+import java.net.URI
+
+@Service
+class TemporalEntityService {
+
+    fun buildTemporalEntity(
+        entityId: URI,
+        attributeAndResultsMap: Map<TemporalEntityAttribute, List<AttributeInstanceResult>>,
+        contexts: List<String>,
+        withTemporalValues: Boolean
+    ): CompactedJsonLdEntity {
+        val temporalAttributes = buildTemporalAttributes(
+            attributeAndResultsMap.filterValues { it.isNotEmpty() },
+            contexts,
+            withTemporalValues
+        )
+
+        return mapOf(
+            "id" to entityId,
+            "type" to JsonLdUtils.compactTerm(attributeAndResultsMap.keys.first().type, contexts)
+        ).plus(temporalAttributes)
+    }
+
+    private fun buildTemporalAttributes(
+        attributeAndResultsMap: Map<TemporalEntityAttribute, List<AttributeInstanceResult>>,
+        contexts: List<String>,
+        withTemporalValues: Boolean
+    ):
+        Map<String, Any> {
+            return if (withTemporalValues) {
+                val attributes = buildAttributesSimplifiedRepresentation(attributeAndResultsMap)
+                mergeAttributeInstancesOnAttributeName(attributes)
+                    .mapKeys { JsonLdUtils.compactTerm(it.key, contexts) }
+            } else
+                mergeAttributeInstancesResultsOnAttributeName(attributeAndResultsMap)
+                    .mapKeys { JsonLdUtils.compactTerm(it.key, contexts) }
+                    .mapValues {
+                        it.value.map { JsonUtils.deserializeObject(it.payload!!) }
```

Comment:
Since for non-simplified representation, we return the content of the payload column, i think we should also migrate datasetIds in the other fix PR.